mesonbuild/programs.py

```diff
@@ -25,14 +25,22 @@
 
 class ExternalProgram(mesonlib.HoldableObject):
 
-    """A program that is found on the system."""
+    """
+    A program that is found on the system.
+    @name: The name of the program
+    @command: Optionally, an argument list constituting the command. Used when
+              you already know the command and do not want to search.
+    @silent: Whether to print messages when initializing
+    @search_dirs: A list of directories to search in first, followed by PATH
+    @exclude_paths: A list of directories to exclude when searching in PATH
```

Comment:
Most of our docstrings use sphinx format, I'd prefer to have one format for all of them

```
"""A program that is found on the system.

:param name: The name of the program.
:param command: ...
```

Reply:
Thanks for the review! Fixed.